crates/c-api/src/module.rs

```diff
@@ -128,3 +128,45 @@ pub extern "C" fn wasm_module_obtain(
         exports,
     }))
 }
+
+#[no_mangle]
+pub extern "C" fn wasmtime_compile_and_serialize(
+    engine: &wasm_engine_t,
+    binary: &wasm_byte_vec_t,
+    ret: &mut wasm_byte_vec_t,
+) -> Option<Box<wasmtime_error_t>> {
+    let mut result = Vec::new();
+    handle_result(
+        compile_and_serialize(&engine.engine, binary.as_slice(), &mut result),
+        |()| {
+            ret.set_buffer(result);
+        },
+    )
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn wasmtime_module_deserialize(
```

Comment:
IIRC the C API already has `wasm_module_{serialize,deserialize}` so could those be implemented instead? (although these may also be good to keep around for better error messages)